pkg/providers/ovirt/mapper/mapper.go

```diff
@@ -0,0 +1,200 @@
+				Spec: kubevirtv1.VirtualMachineInstanceSpec{
+					Domain: kubevirtv1.DomainSpec{},
+				},
+			},
+		},
+	}
+
+	// Map name
+	vmSpec.ObjectMeta.Name, _ = o.resolveVMName(targetVMName, o.vm)
+
+	// Map CPU
+	vmSpec.Spec.Template.Spec.Domain.CPU = o.mapCPU()
+
+	// Map bios
+	vmSpec.Spec.Template.Spec.Domain.Firmware = o.mapFirmware()
+
+	// Map machine type
+	vmSpec.Spec.Template.Spec.Domain.Machine = *o.mapArchitecture()
+
+	return &vmSpec
+}
+
+// MapDisks map the oVirt VM disks to the map of CDI DataVolumes specification, where
+// map id is the id of the oVirt disk
+func (o *OvirtMapper) MapDisks() map[string]cdiv1.DataVolume {
+	diskAttachments, _ := o.vm.DiskAttachments()
+	dvs := make(map[string]cdiv1.DataVolume, len(diskAttachments.Slice()))
+
+	for _, diskAttachment := range diskAttachments.Slice() {
+		attachID, _ := diskAttachment.Id()
+		disk, _ := diskAttachment.Disk()
+		diskID, _ := disk.Id()
+		sdClass := o.getStorageClassForDisk(disk, o.mappings)
+		quantity, _ := resource.ParseQuantity(strconv.FormatInt(disk.MustProvisionedSize(), 10))
+
+		dvs[attachID] = cdiv1.DataVolume{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: cdiAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      attachID,
+				Namespace: o.namespace,
+			},
+			Spec: cdiv1.DataVolumeSpec{
+				Source: cdiv1.DataVolumeSource{
+					Imageio: &cdiv1.DataVolumeSourceImageIO{
+						URL:           o.creds.URL,
+						DiskID:        diskID,
+						SecretRef:     o.creds.SecretName,
+						CertConfigMap: o.creds.ConfigMapName,
+					},
+				},
+				PVC: &corev1.PersistentVolumeClaimSpec{
+					StorageClassName: &sdClass,
+					// TODO: should we have parameter in resourceMapping for this?
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: quantity,
+						},
+					},
+				},
+			},
+		}
+	}
+	return dvs
+}
+
+func (o *OvirtMapper) resolveVMName(targetVMName *string, vm *ovirtsdk.Vm) (string, bool) {
+	if targetVMName != nil {
+		return *targetVMName, false
+	}
+
+	name, ok := vm.Name()
+	if !ok {
+		return "", true
+	}
+
+	name, err := utils.NormalizeName(name)
+	if err != nil {
+		// TODO: should name validation be included in condition ?
+		return "", true
+	}
+
+	return name, false
+}
+
+func (o *OvirtMapper) getStorageClassForDisk(disk *ovirtsdk.Disk, mappings *v2vv1alpha1.OvirtMappings) string {
+	sds, _ := disk.StorageDomains()
+	sd := sds.Slice()[0]
+	for _, mapping := range *mappings.StorageMappings {
+		if mapping.Source.Name != nil {
+			if name, _ := sd.Name(); name == *mapping.Source.Name {
+				return mapping.Target.Name
+			}
+		}
+
+		if mapping.Source.ID != nil {
+			if id, _ := sd.Id(); id == *mapping.Source.ID {
+				return mapping.Target.Name
+			}
+		}
+	}
+
+	return "default" // TODO: Should we use here default, or fail?
```

Comment:
iirc, `default` is just an annotation set for the default storageClass and not an actual name.
If storageClass is not specified, we can should rely on the default storageClass (not sure if it should be provided explicitly or DV can be created without stating it).

By not requiring the field, we can simplify the mappings and the input needed by the user.

Reply:
Default is not needed. If not provided the `default` is used.

Reply:
Done. I've removed the storage class from spec, if not target matches.